compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEControlCodeLowering.cpp

```diff
@@ -354,6 +354,413 @@ struct DmaWaitToTctSyncConverter final
   }
 };
 
+//===----------------------------------------------------------------------===//
+// AIEDeviceBuilder utilities
+//===----------------------------------------------------------------------===//
+// TODO(avarma): Copied from LowerToAIE. Templatize it later because of a few
+//               nuances.
+using BDDimLayoutAndLength = std::pair<AMDAIE::BDDimLayoutArrayAttr, int64_t>;
+
+BDDimLayoutAndLength convertSizeStrideToBDDimLayoutArrayAttr(
```

Comment:
Isn't this function exactly the same, so it can be extracted?

Reply:
This is using AIE specific ops. So I can't extract this, yet. Can be part of future PRs.